packages/sdk/client-services/src/packlets/vault/worker-session.ts

```diff
@@ -48,34 +45,49 @@ export class WorkerSession {
   public bridgeService?: BridgeService;
 
   constructor({
-    getService,
+    serviceHost,
     systemPort,
     appPort,
     shellPort,
+    readySignal,
     options = {
       heartbeatInterval: 1_000
     }
   }: WorkerSessionParams) {
     assert(options);
-    assert(getService);
+    assert(serviceHost);
     this._options = options;
-    this._getService = getService;
-
-    const handlers = {
-      DataService: () => this._getService((services) => services.DataService),
-      DeviceInvitationsService: () => this._getService((services) => services.DeviceInvitationsService),
-      DevicesService: () => this._getService((services) => services.DevicesService),
-      DevtoolsHost: () => this._getService((services) => services.DevtoolsHost),
-      NetworkService: () => this._getService((services) => services.NetworkService),
-      IdentityService: () => this._getService((services) => services.IdentityService),
-      SpaceInvitationsService: () => this._getService((services) => services.SpaceInvitationsService),
-      SpacesService: () => this._getService((services) => services.SpacesService),
-      SystemService: () => this._getService((services) => services.SystemService),
-      TracingService: () => this._getService((services) => services.TracingService)
+    this._serviceHost = serviceHost;
+
+    const middleware: Pick<ClientRpcServerParams, 'handleCall' | 'handleStream'> = {
+      handleCall: async (method, params, handler) => {
+        const error = await readySignal.wait({ timeout: 3_000 });
```

Comment:
could we create a single file inside `@dxos/client` that defines our various timeout lengths and then start to use those everywhere?

Reply:
👍